tests/TestCase/TestSuite/TestCaseTest.php

```diff
@@ -146,6 +146,32 @@ public function testWithErrorReportingWithException(): void
         }
     }
 
+    /**
+     * test deprecated
+     */
+    public function testDeprecated(): void
+    {
+        $this->deprecated(function () {
+            deprecationWarning('test');
+        });
+    }
+
+    /**
+     * test deprecated
+     */
+    public function testDeprecatedWithNoDeprecation(): void
+    {
+        $message = '';
+        try {
+            $this->deprecated(function () {
+            });
+        } catch (\Exception $e) {
+            $message = $e->getMessage();
+        } finally {
+            $this->assertStringStartsWith('Should have at least one deprecation warning', $message);
```

Comment:
You could have this assertion in the catch block and skip finally. You can also use `$this->fail()` at the bottom of the try to ensure an exception was raised.

Reply:
done